src/data_handling/dataloader.py

```diff
@@ -0,0 +1,69 @@
+import json
+import os
+
+import gdown
+import pandas as pd
+
+from src import PROJECT_PATH
+
+
+class DataLoader:
+    def __init__(self,
+                 do_download: bool = False,
+                 folder_link: str = '',
+                 put_data_folder_inside_project_folder: bool = True):
+        """
+        Constructor.
+        :param bool do_download: True if we wish to download data, False if we don't
+        :param str folder_link: The link of the folder containing all necessary data
+        :param bool put_data_folder_inside_project_folder: if True, the generated data folder will
+        be inside the project folder, otherwise it will be the in the folder containing the
+        project folder
+        """
+        self.do_download = do_download
+        self.folder_link = folder_link
+        self.put_data_folder_inside_project_folder = put_data_folder_inside_project_folder
+
+        self.dataset_name = 'time_series_data'
+
+        if self.put_data_folder_inside_project_folder:
```

Comment:
Inkább legyen egy `data_folder_path` argumentum, ami default-ból `None` és azt ellenőrizd, hogy `is None` - ha igen, akkor megy a `PROJECT_PATH/data` mappába, egyébként pedig oda, amit megadnak.